base/client.jl

```diff
@@ -231,9 +231,18 @@ let reqarg = Set(UTF8String["--home",          "-H",
                             "--bind-to"])
     global process_options
     function process_options(opts::JLOptions, args::Vector{UTF8String})
-        if !isempty(args) && (arg = first(args); !isempty(arg) && arg[1] == '-' && in(arg, reqarg))
-            println(STDERR, "julia: option `$arg` is missing an argument")
-            exit(1)
+        if !isempty(args)
+            arg = first(args)
+            if !isempty(arg) && arg[1] == '-' && in(arg, reqarg)
+                println(STDERR, "julia: option `$arg` is missing an argument")
+                exit(1)
+            end
+            idxs = find(x -> x == "--", args)
```

Comment:
what's the rationale for blocking the `--` symbol from user arguments? the usual use of the `--` flag is to permit use of `--` as a filename (or other filenames that look like flags) to be permitted as an unprocessed inputs. as such, it seems odd to me to unconditionally block them.

i would also have expected the following to be valid to allow chaining, where `--` terminates argument parsing for each subsequent parsing of `ARGS`:
```julia
./julia -f -- ./filename.jl -- first args -- user args
```

currently, however, it doesn't get anywhere:
```
$ ./julia -f -- ./filename.jl -- first args -- user args
julia: redundant option terminator `--`
```